### PR TITLE
Fix potential IllegalCapacityException in LLRC when selecting nodes

### PR DESCRIPTION
**Pull request by SivagurunathanV**
Handling LLRC when **live node size < 0**

### CODE REVIEW

**Comment by elasticcla** [comment]
Hi @SivagurunathanV, we have found your signature in our records, but it seems like you have signed with a different e-mail than the one used in yout Git [commit](https://github.com/elastic/elasticsearch/pull/37821.patch). Can you please add both of these e-mails into your Github profile (they can be hidden), so we can match your e-mails to your Github profile?

**Review by cbuescher — CHANGES_REQUESTED** [review]
The change itself looks like the solution proposed in #37739, so that part looks fine. It would be good to have some sort of test though that checks this behaviour. It looks like it should be easy to unit-test RestClient#selectNodes even though it is package-private. There is already a RestClientTests that should be a good place for this.

File: client/rest/src/main/java/org/elasticsearch/client/RestClient.java [hunk 1]
```diff
@@ -398,7 +398,7 @@ private void setHeaders(HttpRequest httpRequest, Collection<Header> requestHeade
         /*
          * Sort the nodes into living and dead lists.
          */
-        List<Node> livingNodes = new ArrayList<>(nodeTuple.nodes.size() - blacklist.size());
+        List<Node> livingNodes = new ArrayList<>(Math.max(0,nodeTuple.nodes.size() - blacklist.size()));
```
Comment (dakrone) [inline client/rest/src/main/java/org/elasticsearch/client/RestClient.java]: Super minor nit, could you add a space after the comma?

```suggestion
        List<Node> livingNodes = new ArrayList<>(Math.max(0, nodeTuple.nodes.size() - blacklist.size()));
```

**Review by dakrone — CHANGES_REQUESTED** [review]
As @cbuescher mentioned, do you think you could add a unit test for this @SivagurunathanV?

File: client/rest/src/main/java/org/elasticsearch/client/RestClient.java (hunk 1 above)
Comment (SivagurunathanV) [reply client/rest/src/main/java/org/elasticsearch/client/RestClient.java]: Done.

**Comment by SivagurunathanV** [comment]
Added test case.

**Comment by dakrone** [comment]
@elasticmachine ok to test

**Comment by SivagurunathanV** [comment]
Hi @dakrone 
I found one test failure issue 

"org.elasticsearch.transport.ConnectTransportException: [node_sm1][127.0.0.1:37453] connect_exception 
Caused by: java.net.ConnectException: Connection refused
on **TcpTransport.java**". 

I would appreciate if can you help to debug on this or any point of reference?

Thanks,
Siva

**Comment by SivagurunathanV** [comment]
Fixed the issue in the build.

**Review by dakrone — APPROVED** [review]

**Review by cbuescher — APPROVED** [review]

**Comment by dakrone** [comment]
Merged this, thanks for working through this @SivagurunathanV!

**Pull request merged**